src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -2443,6 +2443,12 @@ internal BoundExpression GenerateConversionForAssignment(TypeSymbol targetType,
                     GenerateImplicitConversionError(diagnostics, expression.Syntax, conversion, expression, targetType);
                 }
 
+                if (expression.Kind == BoundKind.TupleLiteral)
```

Comment:
>if (expression.Kind == BoundKind.TupleLiteral) [](start = 16, length = 46)

I think we should enter this ```if``` for all valid conversions, see implementation of ```private BoundExpression BindCastCore(ExpressionSyntax node, BoundExpression operand, TypeSymbol targetType, bool wasCompilerGenerated, DiagnosticBag diagnostics)```.